### ISSUE TITLE
use USA address styles in the USA

### ISSUE THREAD

**Issue by TimMagee** [organicmaps/organicmaps#616]
In the USA (and possibly in other places that I don't know about), we write out addresses backwards from the way Europeans do it. Instead of writing "Example Street, 123", we write "123 Example Street".

So I propose that when set to a US locale, addresses be written house number first, followed buy street name instead of street name first like in European (and maybe other) locales.

**Comment by jamescridland** [comment]
I would agree with this, and it's also valid for Canada,
UK, Ireland, New Zealand and Australia.

**Comment by biodranik** [comment]
The idea is good, however, address format should not depend on locale, but on the country where it's located.

**Comment by jamescridland** [comment]
It looks like https://www.grcdi.nl/gsb/world%20address%20formats.html contains the data needed to do this properly; is this required in some form of ini file alongside country codes? Would be happy to compile something.

**Comment by biodranik** [comment]
Yeah, it would be great to make an easily editable DB in some form. But we need to think about how to properly bound it to the current countries/regions list. Maybe even integrate the address data (along with some other useful information) into data/countries.txt? @vng Any ideas?

**Comment by jamescridland** [comment]
![Screenshot_20210710-185344](https://user-images.githubusercontent.com/231941/125157801-4c910c00-e1b0-11eb-830f-f2a8eafb21df.jpg)

Just so we're clear of the use-case here: the second result in this screen, which is https://www.openstreetmap.org/node/7412956206 , should ideally say...

```
TOMRA
Recycling Centre
9 Counihan Road, Seventeen Mile Rocks QLD
```

i.e.

```
**(:name)**
(:place type)
(:addr:housenumber) (addr:street), (addr:suburb) (addr:state)
```

It currently says "Brisbane City" at the front, which is incorrect (it's a fifteen mile drive from Brisbane City); the suburb isn't mentioned, and nor is the state. For a search result, it's important to help disambiguate between this branch and the one in "Brisbane City, Brisbane, Australia" which is [this one](https://www.openstreetmap.org/node/6795260067), correctly listed as

```
TOMRA
Recycling Centre
Montague Road, West End QLD
```

You'll also note that "Recycling Centre" should be spelt this way, in English (Australian), rather than "Recycling Center" (which is US English and incorrect).

**Comment by biodranik** [comment]
Please create a separate issue for non-US English interface translations (type name is translated as a part of the interface).

Also, a separate issue is the incorrect suburb/city detection. It's not fully implemented yet.

Here let's focus on the proper address formatting.

**Comment by jamescridland** [comment]
Have done - #829 - for the US English vs non-US English. Thanks.

Let's leave the suburb/city bit for now: it's less jarring than the "Downing Street, 10" :)

**Comment by natrius** [comment]
As an idea: Ask on startup with drag&drop buttons with a default (for europe maybe)

```
Please choose on how you enter you adress:
(addr:street) (addr:housenumber), (addr:suburb)/(addr:city) (addr:state)
```

With the posibility to drag it around so anybody would be able to enter the adress as he likes.

**Comment by jamescridland** [comment]
This doesn't need to be a user-definable setting: it's a cultural standard.
Personally, I suspect that this is most easily done by using the user settings for the device language. 

en is all (addr:housenumber) (addr:street), (addr:suburb), (addr:city) (addr:state)
fr is all (addr:street) (addr:housenumber), (addr:suburb), (addr:city) (addr:state)

Canada is the obvious one here that we need to take care with. I'm checking with a friendly French Canadian, to understand whether this should be set on device language (en-CA vs fr-CA).

If French Canadians would say "Bleu Rue 24" and not "24 Blue Road" then I think it should be relatively easy to include as a setting based on the device language.

**Comment by biodranik** [comment]
Good idea. Also, in the UK people search using post codes.

**Comment by jamescridland** [comment]
Just to come back on my suggestion above... having spoken to French Canadians, they're similar to French French... but to correct what I said above...

en is all (addr:housenumber) (addr:roadname) (addr:roadtype), (addr:suburb), (addr:city) (addr:state)
fr is all (addr:housenumber) (addr:roadtype) (addr:roadname) (addr:suburb), (addr:city) (addr:state)

So, as an Englishman, I'd say "25 Blue Road", but as a Frenchman, I'd say "25 Rue Bleu".

**Comment by biodranik** [comment]
Thanks for the updates! We display streets as they are named in OSM. There are some road types synonyms though in the search engine: https://github.com/organicmaps/organicmaps/blob/99c5b878a048b467d1c66cf5a5ff0ce04975c73b/indexer/search_string_utils.cpp#L352

**Comment by natrius** [comment]
Just to be clear to everyone not contributing to OpenStreetMap: In osm adresses are added with a `addr:[tag]=[value]` as seen in https://wiki.openstreetmap.org/wiki/Key:addr and the roadtype is probably included in `addr:street` already. But i'm pretty sure the developers are aware of that. 

Still, i like the default but i'm sure sometimes it would be good to be able to change the 'default' way. Because people save the adress in contacts freestyle anyway and so they don't have to switch their prefered adress-saving-style so it does not interfere with other apps. (Because they started to save it intentionally wrong for another app. osmand was long not able to use the adress-scheme used in my country - i just never used it)

**Comment by jamescridland** [comment]
@biodranik "We display streets as they are named in OSM."

I think there's some confusion here - partly because this has become a messy report. Sorry. My fault.

![Screen Shot 2021-07-26 at 7 32 40 pm](https://user-images.githubusercontent.com/231941/126967326-012f77cb-2e2b-4353-a9ed-e1b41058b380.png)

...is the output for [this query](https://nominatim.openstreetmap.org/search?q=10+downing+street,+London&addressdetails=1&format=json).

Organic Maps would display this as "Downing Street 10".

It is "10 Downing Street".

**Comment by biodranik** [comment]
@jamescridland I answered above to that question:

> how do you display a road in Paris to someone who is using an en-US phone? I can't imagine that an English person sees "25 St Michel Rue" when a French person sees "25 Rue St Michel".

It was not about the house number ordering.

@natrius we discuss displaying, not editing, so OSM should be safe here :)

**Comment by TheAdventurer64** [comment]
Here's the way I think we should approach the issue: We should base it on what country you're in. For example, if you're in the United States, the address should be formatted as "123 Example Street". If you're in the United Kingdom,addresses should be formatted as "Example Street, 123". This is the way that software like the iD editor and Google Maps do addresses, and I believe that Organic Maps can do the same.

**Comment by biodranik** [comment]
@TheAdventurer64 What if I travel to the US from the EU and check/search for some place in the US? Each country should have its own format.

**Comment by TheAdventurer64** [comment]
> @TheAdventurer64 What if I travel to the US from the EU and check/search for some place in the US? Each country should have its own format.

My bad - I should've specified - the address formatting should be based on which country you're searching in. If you're searching for places that are within the US, the address should be "123 Street". If you're searching for places located inside the UK, addresses should be formatted to "Street, 123".

**Comment by jamescridland** [comment]
Agree with setting this by country, not by user locale.

But the UK is 12 High Street. Not High Street, 12.



https://james.cridland.net • https://podnews.net

**Comment by RedAuburn** [comment]
Ideally we should integrate https://github.com/OpenCageData/address-formatting/

**Comment by AndydeCleyre** [comment]
In addition to the order of the elements and the lack of zip code in result items being confusing, here in New York City it's essential to specify which "borough" the address is in, when outside Manhattan. We use it (e.g. Queens, Brooklyn) in place of the city name, when listing an address.

Would that fall under this issue as well?

**Comment by deedend** [comment]
Same issue in Australia, hard to distinguish the suburb in my case:
![Screenshot_2024-07-20-09-32-09-202-edit_app organicmaps](https://github.com/user-attachments/assets/9b394ebd-bfb3-4fde-aba3-ef346e8947fa)

**Comment by RedAuburn** [comment]
for this, imo we need a c++ implementation of https://github.com/OpenCageData/address-formatting/

**Comment by pastk** [comment]
> Same issue in Australia, hard to distinguish the suburb in my case:

This is a separate issue:
- https://github.com/organicmaps/organicmaps/issues/6784
